pymongo/asynchronous/periodic_executor.py

```diff
@@ -65,7 +65,21 @@ def __repr__(self) -> str:
         return f"<{self.__class__.__name__}(name={self._name}) object at 0x{id(self):x}>"
 
     def _run_async(self) -> None:
-        asyncio.run(self._run())  # type: ignore[func-returns-value]
+        # The default asyncio loop implementation on Windows
+        # has issues with sharing sockets across loops (https://github.com/python/cpython/issues/122240)
+        # We explicitly use a different loop implementation here to prevent that issue
+        if (
+            not _IS_SYNC
+            and sys.platform == "win32"
+            and asyncio.get_event_loop_policy() == asyncio.WindowsProactorEventLoopPolicy
+        ):
+            loop = asyncio.WindowsSelectorEventLoopPolicy().new_event_loop()  # type: ignore[attr-defined]
```

Comment:
How about we just always use WindowsSelectorEventLoopPolicy on windows for now. I'm  worried the WindowsProactorEventLoopPolicy check won't work for custom policies that use  the proactor loop.

Reply:
Let's also call asyncio.SelectorEventLoop() directly.